Snakefile

```diff
@@ -452,3 +452,26 @@ rule map:
     shell:
         "scripts/map.py {input.img} {input.labels} {output} {input.predicts}"
 
+### extract image rule: for recovering the source images given a set of labels
+rule extract_images:
+    """for recovering the source images given a set of labels from segment_map(_exp).tiff"""
+    input:
+        sourceDir =  config['out']+"/{sample}/rev_transforms/",
```

Comment:
```suggestion
        sourceDir =  config['out']+"/{sample}/rev_transforms/"
```
I've sometimes had problems executing Snakemake pipelines when there's a trailing comma at the end of the input. But it might not be giving you issues